qchem/pennylane_qchem/qchem/obs.py

```diff
@@ -340,9 +340,73 @@ def get_spinZ_matrix_elements(mol_name, hf_data, n_active_electrons=None, n_acti
     return spinz_matrix_elements
 
 
+def get_particle_number_table(mol_name, hf_data, n_active_electrons=None, n_active_orbitals=None):
```

Comment:
For the particular case of the particle number operator is difficult to talk about matrix elements since in second quantization N is defined straightforward. I have been more explicit on this point in the docstring. It was easier to clarified this for S^2 and S_z.

Reply:
I still think it's better to use `elements` for consistency in the `obs` module. Otherwise it may seem that this function is doing something different than for the spin observable, even though it isn't really

Reply:
It might also be good to consider the use of 'get'. To me it seems a bit superfluous --- it doesn't really add anything descriptive. Something like `load_particle_number_elements`, however, straight away indicates that this function is loading data from a file.

I think, to be honest, the main issue with naming this function is that it is trying to do two things:

* Loading the molecular data from a file

* Computing the particle number elements

Does it make more sense to have these function input the active space directly, rather than the file location?

```python
active_space = qml.qchem.active_space(mol_name, hf_data)
pn_table, pn_docc = particle_number_elements(*active_space)
```

This has the advantages that:

* It is _much_ more clear what is happening from the function names; `particle_number_elements` simply computes the particle number elements, nothing more.

* The file only needs to be loaded once, and multiple computations can be performed on the data. For large molecules with big file sizes, I/O can be a bit bottleneck if every observable function is re-reading the file from disk.

In fact, it might be worth re-thinking this across `qchem.obs` --- does every user-facing function need to re-import the molecular data. For instance, we could separate loading of molecular data into a function that returns some sort of data-structure, and use this as input to the various computations:

```python
mol_data = qml.qchem.load_molecule(mol_name, hf_data)
active_space = qml.qchem.active_space(mol_data)
pn_table, pn_docc = particle_number_elements(mol_data)
```